create_derivatives/routines.py

```diff
@@ -68,8 +69,61 @@ class JP2Maker:
 
 
 class PDFMaker:
-    # TO DO: make PDF derivates, compress, OCR
-    pass
+    """Creates concatenated PDF file from JP2 derivatives.
+
+    Creates PDF directory in bag's data directory, creates PDF, then compresses and OCRs the PDF
+
+    Returns:
+        A tuple containing human-readable message along with list of bag identifiers.
+        Exceptions are raised for errors along the way.
```

Comment:
The returns are just the tuple and the message? The pdfs that are created are not returns, right? Asking for my own reference, not challenging this.

Reply:
Correct, what's being returned is a message and a list of bag identifiers. The list of bag identifiers is created  in line 83 and added to in line 91.